src/main/java/com/serjltt/moshi/adapters/LastElement.java

```diff
@@ -0,0 +1,65 @@
+package com.serjltt.moshi.adapters;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.JsonQualifier;
+import com.squareup.moshi.Moshi;
+import com.squareup.moshi.Types;
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.Type;
+import java.util.Set;
+
+/**
+ * Indicates that the annotated type/field is <strong>expected</strong> to be the last element of a
+ * json array.
+ *
+ * <p>For example if a json object is returned as:
+ * <pre>
+ *   [
+ *    {
```

Comment:
Well, you can use if for the last element not only in array, but in object case also. It's totally the same like in firstElement.